opsdroid/connector/matrix/connector.py

```diff
@@ -36,11 +36,20 @@ def __init__(self, config, opsdroid=None):  # noqa: D107
         self.mxid = config['mxid']
         self.nick = config.get('nick', None)
         self.homeserver = config.get('homeserver', "https://matrix.org")
-        self.password = config['password']
         self.room_specific_nicks = config.get("room_specific_nicks", False)
         self.session = None
         self.filter_id = None
         self.connection = None
+        try:
```

Comment:
Rather than this `try / except` block I would just `.get()` on the dict like this:
```python
self.token = config.get("token", None)
self.password = config.get("password", None)
```